File: Selenium/Project/CRM_Activity7.java
package crm_project;

import org.testng.annotations.Test;
import org.testng.annotations.BeforeClass;

import static org.testng.Assert.assertEquals;
import static org.testng.Assert.assertTrue;

import java.time.Duration;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.interactions.Actions;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.testng.Assert;
import org.testng.annotations.AfterClass;

public class CRM_Activity7 {
 
//declare WebDriver 
		WebDriver driver;
		
	  
	  @BeforeClass
	  public void setUp() {
		  //initialize webdriver
		  driver = new ChromeDriver();
	
		  //open the page
		  driver.get("http://alchemy.hguy.co/crm");
	  }
	 
	  @Test(priority = 1)
	  public void pageLoginTest() {
		  //find the username and type in the username
		  driver.findElement(By.id("user_name")).sendKeys("admin");
		  //find the passwrd filed and type in the passwrd
		  driver.findElement(By.id("username_password")).sendKeys("pa$$w0rd");
		  
		  
		  //find the Log in button and click it
		  driver.findElement(By.xpath("//input[@id='bigbutton']")).click();
		  String homePageUrl=driver.getCurrentUrl();
		  Assert.assertTrue(homePageUrl.contains("Home"), "Login failed");
	  }
	  @Test(priority=2)
	  public void salesItemTest() throws InterruptedException {
		  driver.manage().window().maximize();
		 
		  //find the sales item and click it
		
		driver.findElement(By.xpath("//a[text()='Sales']")).click();
		 //find the leads item and click it
	 driver.findElement(By.xpath("//a[text()='Leads']")).click();
		
		 Thread.sleep(1000);
		 // find  the  additional details icon from the table
	     driver.findElement(By.xpath("(//span[(@class='suitepicon suitepicon-action-info')])[1]")).click();
	     Thread.sleep(1000);
		
	     // find the mobile number on popup and print it
	     WebElement popupElement= driver.findElement(By.xpath("//span[@class='phone']"));
	     String mobile=popupElement.getText();
	
	      
	      System.out.println("Mobile Number is:"+ mobile);
	      Assert.assertNotNull(mobile,"Mobile number cannot be null");
		
	  }
	  @AfterClass
	  public void tearDown() {
	  //quit the browser
	  driver.quit();
	  }
	


}

File: Selenium/Project/crm_activity7.py
# Import webdriver from selenium
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.color import Color
from selenium.webdriver.support import expected_conditions as EC
from selenium.webdriver.support.wait import WebDriverWait
import time
# Start the Driver
driver=webdriver.Chrome()
 
    # Navigate to the URL
driver.get("http://alchemy.hguy.co/crm")
#find the username and type in the username
driver.find_element(By.ID,"user_name").send_keys("admin")
  #find the passwrd filed and type in the passwrd
driver.find_element(By.ID,"username_password").send_keys("pa$$w0rd")
			  
			  
# find the Log in button and click it
driver.find_element(By.XPATH,"//input[@id='bigbutton']").click()
homePage=driver.find_element(By.ID,"tab0").text
if(homePage== "SUITECRM DASHBOARD"):
    print(" homepage has opened")
else:
    print("Login failed")
#find the navigation menu
driver.fullscreen_window()
#find the sales item and click it
		
driver.find_element(By.XPATH,"//a[text()='Sales']").click()
#find the leads item and click it
driver.find_element(By.XPATH,"//a[text()='Leads']").click()
		
time.sleep(2)
 #find  the  additional details icon from the table
driver.find_element(By.XPATH,"(//span[(@class='suitepicon suitepicon-action-info')])[1]").click()
time.sleep(2)
		
 #find the mobile number on popup and print it 
popup_element= driver.find_element(By.XPATH,"//span[@class='phone']")
mobile=popup_element.text
	
	      
print("Mobile Number is:"+ mobile)
	     